### PR TITLE
Replace data csv with sim

### PR DESCRIPTION
**Pull request by akoen**
Allows us to spoof sensor data over SIM.

The legacy DataSpoof class was renamed to FileDataSpoof if we need it in the future.

See also  https://github.com/UBC-Rocket/UBCRocketGroundStation/pull/37

### CODE REVIEW

File: avionics/envs/x86/include/sensors/barometer.h [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include "../env_config.h"
 #include "../sensor_data.hpp"
 #include "sensors_interface.h"
+#include <bits/stdint-uintn.h>
```
Comment (jesseli2002) [inline avionics/envs/x86/include/sensors/barometer.h]: bits is a GCC specific header - should instead use [`cstdint`](https://en.cppreference.com/w/cpp/header/cstdint)

File: avionics/envs/x86/include/stdio_controller.hpp [hunk 2]
```diff
@@ -1,12 +1,14 @@
 #pragma once
 
 #include <atomic>
+#include <bits/stdint-uintn.h>
```
Comment (jesseli2002) [inline avionics/envs/x86/include/stdio_controller.hpp]: [see above on bits]

File: avionics/envs/x86/include/sensors/barometer.h [hunk 3]
```diff
@@ -21,5 +22,6 @@ class Barometer : public SensorBase<2> {
     SensorStatus getStatus() { return SensorStatus::NOMINAL; }
 
   private:
-    DataSpoof<2> dat{BAROMETER_DATA, data_};
```
Comment (jesseli2002) [inline avionics/envs/x86/include/sensors/barometer.h:24]: `BAROMETER_DATA` doesn't seem to have any use anymore; if the definition hasn't been removed make sure to do so

File: avionics/envs/x86/include/stdio_controller.hpp [hunk 4]
```diff
@@ -91,28 +139,57 @@ class StdIoController {
      * @return Read value.
      */
     static int requestAnalogRead(uint8_t const pin_id) {
-        run_input_ = false; // input_ will now run at most once more
-        putPacket('a', &pin_id,
-                  1);  // ensures that at least one more SIM packet exists
-        input_.join(); // so this won't block
+        uint8_t packet_id = static_cast<uint8_t>(PacketIds::analog_read);
```
Comment (jesseli2002) [inline avionics/envs/x86/include/stdio_controller.hpp]: I wonder if constexpr will work here, just to indicate that this variable isn't going to change (and/or ALL_CAPS to indicate constant)

File: avionics/envs/x86/include/stdio_controller.hpp [hunk 5]
```diff
@@ -91,28 +139,57 @@ class StdIoController {
...
+
   private:
+    /**
+     * @brief reinterprets an array of 4 chars to a floating point
+     * number.
+     * @param data: an array of 4 chars.
+     * @return the floating point representation
+     */
+
+    // NOTE Be mindful of endianess
```
Comment (jesseli2002) [inline avionics/envs/x86/include/stdio_controller.hpp]: nitpicking but I believe that automatically generated documentation assumes that the comment block comes immediately before the function definition (not that we're generating documentation) - maybe move this comment inside

although for that matter, this should be clarified to say that ground station is taking care of endianness

File: avionics/envs/x86/include/stdio_controller.hpp [hunk 6]
```diff
@@ -91,28 +139,57 @@ class StdIoController {
...
+    static float charsToFloat(uint8_t data[4]) {
+        float f;
+        std::memcpy(&f, data, sizeof(float));
```
Comment (jesseli2002) [inline avionics/envs/x86/include/stdio_controller.hpp]: optionally use the FLOAT_SIZE variable defined earlier

**Review by jesseli2002 — CHANGES_REQUESTED** [review]
Mostly nitpicking small things.

**Comment by SilentSys** [comment]
[Request sensor spoofing through SIM](https://trello.com/c/SGMhyk9K/239-request-sensor-spoofing-through-sim)

File: avionics/.vscode/settings.json [hunk 7]
```diff
@@ -1,81 +0,0 @@
-{
```
Comment (SilentSys) [inline avionics/.vscode/settings.json:1]: Should add .vscode to gitignore now that its removed so that git doesnt try to track it every time.

**Review by SilentSys — CHANGES_REQUESTED** [review]
Looks like GPS sim was enabled in the SW but its not done in here. Can we please add the DataSpoof to the GPS as well just to keep both repos consistent with each other?

File: avionics/.vscode/settings.json (hunk 7 above)
Comment (akoen) [reply avionics/.vscode/settings.json:1]: Done

**Review by SilentSys — APPROVED** [review]
Ran some tests, works like a charm!

**Comment by SilentSys** [comment]
[Remove .vscode from the repository](https://trello.com/c/bBcUN9yR/251-remove-vscode-from-the-repository)

**Review by jesseli2002 — APPROVED** [review]

**Pull request merged**